bin/propolis-server/src/lib/spec/builder.rs

```diff
@@ -59,6 +62,10 @@ pub(crate) enum SpecBuilderError {
     #[error("pvpanic device already specified")]
     PvpanicInUse,
 
+    #[cfg(not(feature = "omicron-build"))]
+    #[error("migration failure injection already enabled")]
+    MigrationFailureInjectionInUse,
```

Comment:
Yeah, I think we can keep this as-is. Since all this is internal to the server anyway we can always change our minds if we end up with a truly unwieldy number of singleton components.

Reply:
Thanks for bringing this up, though! I hadn't even considered the other approach.